File: calculator.nim
import strutils

proc calculate(op: char, num1, num2: float): float =
  case op:
    of '+': return num1 + num2
    of '-': return num1 - num2
    of '*': return num1 * num2
    of '/':
      if num2 != 0:
        return num1 / num2
      else:
        echo "Error: Division by zero"
        return 0.0
    else:
      echo "Error: Unsupported operation"
      return 0.0

proc main() =
  echo "Simple Calculator"
  echo "Enter first number: "
  let input1 = readLine(stdin)
  let num1 = input1.parseFloat()

  echo "Enter operation (+, -, *, /): "
  let operation = readChar(stdin)

  echo "Enter second number: "
  let input2 = readLine(stdin)
  let num2 = input2.parseFloat()

  let result = calculate(operation, num1, num2)
  echo "Result: ", result

main()

File: calculator.ts
import * as readline from 'readline';

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function calculate(op: string, num1: number, num2: number): number | string {
    switch (op) {
        case '+':
            return num1 + num2;
        case '-':
            return num1 - num2;
        case '*':
            return num1 * num2;
        case '/':
            if (num2 !== 0) {
                return num1 / num2;
            } else {
                return 'Error: Division by zero';
            }
        default:
            return 'Error: Unsupported operation';
    }
}

function askQuestion(query: string): Promise<string> {
    return new Promise(resolve => rl.question(query, resolve));
}

async function main() {
    console.log("Simple Calculator");

    const input1 = await askQuestion("Enter first number: ");
    const num1 = parseFloat(input1);

    const operation = await askQuestion("Enter operation (+, -, *, /): ");

    const input2 = await askQuestion("Enter second number: ");
    const num2 = parseFloat(input2);

    const result = calculate(operation, num1, num2);
    console.log("Result:", result);

    rl.close();
}

main().catch(err => {
    console.error(err);
    rl.close();
});
